src/program.cpp

```diff
@@ -1521,6 +1552,26 @@ cl_int cvk_entry_point::init() {
         m_pod_buffer_size = round_up(m_pod_buffer_size, 4);
     }
 
+    // Take the size of image metadata into account for the pod buffer size
+    bool image_md_is_max = false;
+    for (auto md : m_image_channel_metadata) {
+        auto order_offset = md.second.first;
+        auto data_type_offset = md.second.second;
+        if (order_offset != IMAGE_METADATA_EMPTY &&
+            order_offset > m_pod_buffer_size) {
+            m_pod_buffer_size = order_offset;
+            image_md_is_max = true;
+        }
+        if (data_type_offset != IMAGE_METADATA_EMPTY &&
+            data_type_offset > m_pod_buffer_size) {
+            m_pod_buffer_size = data_type_offset;
+            image_md_is_max = true;
+        }
+    }
```

Comment:
Not sure I understand. Does this assume that the offset into the push constants provided by clspv can be used as an offset into the POD buffer? Surely this won't always hold. Maybe a test case for when this doesn't hold would be useful?

Reply:
Yes, that's a first basic implementation which is only capable of sending those values through the pod. And yes it was not checking for the pod buffer size. It is now checking it and return an error if it exceeds it.